### ISSUE TITLE
Use Transformer encoder decoder backbone

### ISSUE THREAD

**Issue by lifengheng** [CyberAgentAILab/layout-dm#25]
Hi, I notice layoutDM use Transformer encoder only backbone for faster generation at the sacrifice of quality in your paper.  I want to know how much improvement can be achieved if layoutDM use Transformer encoder decoder backbone. Did you do some similar experiments before? I tried to add torch.nn.TransformerDecoder to layoutDM but the loss went wrong. Is layoutDM decoder different from nn.TransformerDecoder?

**Comment by naoto0804** [comment]
Hi, using a Transformer encoder is **not** for faster generation. It is because the number of fields to be generated is fixed (c.f. a BERT-like model). In that case, it is natural to use a Transformer encoder.

**Comment by lifengheng** [comment]
Sorry, I got a wrong understanding of the encoder. Thanks for your reply.

**Comment by naoto0804** [comment]
Happy to hear that you get it right!
